db/upgrade.php

```diff
@@ -565,5 +565,31 @@ function xmldb_local_recompletion_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2021112400, 'local', 'recompletion');
     }
 
+    if ($oldversion < 2023022100) {
+
+        // Define table local_recompletion_cha to be created.
+        $table = new xmldb_table('local_recompletion_cha');
+
+        // Adding fields to table local_recompletion_cha.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('choiceid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('userid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
+        $table->add_field('optionid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
+        $table->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
+        $table->add_field('course', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
+
+        // Adding keys to table local_recompletion_cha.
+        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        $table->add_key('choiceid', XMLDB_KEY_FOREIGN, ['choiceid'], 'choice', ['id']);
```

Comment:
this key (choiceid) isn't defined in the install.xml file - might be easier to remove it?

Reply:
looks like you're rolling this on your own rather than using xmldb to create these chunks..... I see at least one difference in the info that is created in the upgrade.php file versus what is defined in install.xml.... rather than point that specific instance out... any chance you could get the create table script from the xmldb editor so that it completely matches properly?

Reply:
Thanks @danmarsden , you are right: I was not using xmldb editor, as I didn't know it. I have now created the files with the xmldb editor, hopefully the will be fine now. Sorry for wasting your time with this and thanks again for your help.